homeassistant/components/lektrico/binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Support for Lektrico binary sensors entities."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import ATTR_SERIAL_NUMBER, CONF_TYPE, EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LektricoConfigEntry, LektricoDeviceDataUpdateCoordinator
+from .entity import LektricoEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class LektricoBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Lektrico binary sensor entity."""
+
+    value_fn: Callable[[dict[str, Any]], bool]
+    attributes_fn: Callable[[Any], dict[Any, bool]] | None = None
+
+
+BINARY_SENSORS: tuple[LektricoBinarySensorEntityDescription, ...] = (
+    LektricoBinarySensorEntityDescription(
+        key="errors",
+        translation_key="errors",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        value_fn=lambda data: bool(data["has_active_errors"]),
+        attributes_fn=lambda data: {
+            "state_e_activated": bool(data["state_e_activated"]),
+            "overtemp": bool(data["overtemp"]),
+            "critical_temp": bool(data["critical_temp"]),
+            "overcurrent": bool(data["overcurrent"]),
+            "meter_fault": bool(data["meter_fault"]),
+            "undervoltage_error": bool(data["undervoltage_error"]),
+            "overvoltage_error": bool(data["overvoltage_error"]),
+            "rcd_error": bool(data["rcd_error"]),
+            "cp_diode_failure": bool(data["cp_diode_failure"]),
+            "contactor_failure": bool(data["contactor_failure"]),
+        },
```

Comment:
We should have individual sensors for this, rather than having it in the attributes, as having this as separate entities will improve how users can automate with this

Reply:
I made individual binary sensors, thank you.